engine/src/main/java/io/zeebe/engine/processor/StreamProcessor.java

```diff
@@ -154,6 +162,26 @@ protected void onActorCloseRequested() {
     return closeFuture;
   }
 
+  @Override
+  protected void handleFailure(final Exception failure) {
+    if (this.failureListener != null) {
+      this.failureListener.onFailure();
+    }
+    actor.fail();
+  }
+
+  @Override
+  public void onActorFailed() {
+    phase = Phase.FAILED;
```

Comment:
Which other resources?

Reply:
Idk things like this:


```
  @Override
  protected void onActorClosing() {
    processingContext.getLogStreamReader().close();

    if (onCommitPositionUpdatedCondition != null) {
      logStream.removeOnCommitPositionUpdatedCondition(onCommitPositionUpdatedCondition);
      onCommitPositionUpdatedCondition = null;
    }
  }

  @Override
  protected void onActorClosed() {
    closeFuture.complete(null);
    LOG.debug("Closed stream processor controller {}.", getName());
  }

  @Override
  protected void onActorCloseRequested() {
    if (!isFailed()) {
      lifecycleAwareListeners.forEach(StreamProcessorLifecycleAware::onClose);
    }
  }
```

Also should we introduce a new listener methods for the lifecycle listeners?

Reply:
:see_no_evil: my bad. Doing cleanup onActorFailed is always same as the steps in normal close. Let me see if we can do it any better here to avoid such mistakes in future.

Reply:
> Also should we introduce a new listener methods for the lifecycle listeners?
May be. But is it useful? We can treat "FAILED" phase similar to closed, as we are doing now.

Reply:
Ya. It would make sense to add a new listener method, or atleast make sure that onClose listener is invoked when failed. But adding a new listener method would be meaningful.